csharp-intermediate-topics/HowToConvertAStreamToAFileInCSharp/Tests/ConvertAStreamToAFileInCSharpTests.cs

```diff
@@ -0,0 +1,80 @@
+using System.Reflection;
+using HowToConvertAStreamToAFileInCSharp;
+
+namespace Tests;
+
+[TestClass]
+public class ConvertAStreamToAFileInCSharpTests
+{
+
+    [TestMethod]
+    public void WhenCopyToFile_FileExists()
+    {
+        var destinationPath = GetPath();
+        var memoryStream = GetMemoryStream();
+        ConvertAStreamToAFileInCSharp.CopyToFile(memoryStream, Path.Combine(destinationPath, @"CopyTo.png"));
+
+        Assert.IsTrue(File.Exists(Path.Combine(destinationPath, @"CopyTo.png")));
+    }
+
+    [TestMethod]
+    public void WhenWriteToFile_FileExists()
+    {
+        var destinationPath = GetPath();
+        var memoryStream = GetMemoryStream();
+        ConvertAStreamToAFileInCSharp.WriteToFileStream(memoryStream, Path.Combine(destinationPath, @"WriteTo.png"));
+
+        Assert.IsTrue(File.Exists(Path.Combine(destinationPath, @"WriteTo.png")));
+    }
+
+    [TestMethod]
+    public void WhenWriteByteToFile_FileExists()
+    {
+        var destinationPath = GetPath();
+        var memoryStream = GetMemoryStream();
+        ConvertAStreamToAFileInCSharp.WriteByteToFileStream(memoryStream, Path.Combine(destinationPath, @"WriteByte.png"));
+
+        Assert.IsTrue(File.Exists(Path.Combine(destinationPath, @"WriteByte.png")));
+    }
+
+    [TestMethod]
+    public void WhenWriteAllBytesToFile_FileExists()
+    {
+        var destinationPath = GetPath();
+        var memoryStream = GetMemoryStream();
+        ConvertAStreamToAFileInCSharp.WriteAllBytesFile(memoryStream, Path.Combine(destinationPath, @"WriteAllBytes.png"));
+
+        Assert.IsTrue(File.Exists(Path.Combine(destinationPath, @"WriteAllBytes.png")));
+    }
+
+    private string GetPath()
+    {
+        var directory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        if (!Directory.Exists(directory))
+        {
+            return string.Empty;
+        }
+
+        var sourceFile = Path.Combine(directory, @"Files\source.png");
+        var destinationPath = Path.Combine(directory, @"Files\");
```

Comment:
@bozo-spoljaric - don't include the file separator in these methods, `Path.Combine()` will add the system-appropriate separator for us. Also, since we are not returning the source file, there is no need to compute it here, we can just compute the destination path.

Reply:
Did so, thank you